### PR TITLE
pupx: modify for wayland

### PR DESCRIPTION
**Pull request by 01micko**
add killall swayidle and restart swayidle

### CODE REVIEW

File: woof-code/rootfs-skeleton/usr/bin/pupx [hunk 1]
```diff
@@ -133,9 +137,15 @@ eval "$RETPARAMS"
 if [ -f ~/.Xwaylandrc ]; then
     if [ "$SCREENSAVERFLAG" = "true" -a "$SCREENSAVERDELAY" -gt 0 ]; then
         sed -i "s%^XWAYLAND_SCREENSAVER_DELAY=.*%XWAYLAND_SCREENSAVER_DELAY=$SCREENSAVERDELAY%" ~/.Xwaylandrc
+		if pidof labwc >/dev/null 2>&1 ; then # could add sway here
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/pupx:140]: In a Puppy with labwc but without Cage, ~/.Xwaylandrc doesn't exist

Comment (01micko) [reply woof-code/rootfs-skeleton/usr/bin/pupx:140]: Cross that bridge when we come to it. I _was_ running pure wayland but hardly anything worked so call it a transitional phase.

Comment (01micko) [reply woof-code/rootfs-skeleton/usr/bin/pupx:140]: However, if it's done now (as in #2689 ) at least we are somewhat future proof.

**Review by dimkr — APPROVED** [review]

File: woof-code/rootfs-skeleton/usr/bin/pupx (hunk 1 above)
Comment (01micko) [reply woof-code/rootfs-skeleton/usr/bin/pupx:140]: Done in fd27bb4

File: woof-code/rootfs-skeleton/usr/bin/pupx [hunk 2]
```diff
@@ -146,6 +146,12 @@ if [ -f ~/.Xwaylandrc ]; then
         sed -i "s%^XWAYLAND_SCREENSAVER_DELAY=.*%XWAYLAND_SCREENSAVER_DELAY=%" ~/.Xwaylandrc
 	fi
     exit
+elif [ -f "$XDG_CONFIG_HOME/swayidle/config" ]; then # created by set_wl_monsleep at startup absent
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/pupx:149]: WAYLAND_DISPLAY is set when running labwc, so you'll never reach this `elif` - is that right?

File: woof-code/rootfs-skeleton/usr/bin/pupx [hunk 3]
```diff
@@ -4,7 +4,7 @@ SCREENSAVERFLAG='yes'
 SCREENVISIBLE="enabled"
 
 if [ -n "$WAYLAND_DISPLAY" ]; then
-    . ~/.Xwaylandrc
+    [ -f .Xwaylandrc ] && . ~/.Xwaylandrc
```
Comment (01micko) [inline woof-code/rootfs-skeleton/usr/bin/pupx:7]: Depends on existence of `~/.Xwaylandrc`

File: woof-code/rootfs-skeleton/usr/bin/pupx (hunk 2 above)
Comment (01micko) [reply woof-code/rootfs-skeleton/usr/bin/pupx:149]: See line 7 of the diff

Comment (01micko) [reply woof-code/rootfs-skeleton/usr/bin/pupx:149]: BTW< I tested with replacing every instance of `,Xwaylandrc` with a dummy and it works as expected.. you get an unchecked checkbox which you need to check to sensitize and change the spinbutton value.

File: woof-code/rootfs-skeleton/usr/bin/pupx [hunk 4]
```diff
@@ -146,6 +146,12 @@ if [ -f ~/.Xwaylandrc ]; then
         sed -i "s%^XWAYLAND_SCREENSAVER_DELAY=.*%XWAYLAND_SCREENSAVER_DELAY=%" ~/.Xwaylandrc
 	fi
     exit
+elif [ -f "$XDG_CONFIG_HOME/swayidle/config" ]; then # created by set_wl_monsleep at startup absent
+	CURDELAY=$(grep -o 'timeout [0-9]*[0-9]' $XDG_CONFIG_HOME/swayidle/config)
```
Comment (01micko) [inline woof-code/rootfs-skeleton/usr/bin/pupx:150]: Only thing I need to do is check for stupid values here, like a single digit.

OR

Perhaps the range should start at something more sane like a min of `30` (which is still silly, but ok for testing).

**Pull request merged**